pkg/detector/policy.go

```diff
@@ -69,7 +69,7 @@ func (d *ResourceDetector) propagateResource(object *unstructured.Unstructured,
 		}
 		d.RemoveWaiting(objectKey)
 		metrics.ObserveFindMatchedPolicyLatency(start)
-		return d.ApplyPolicy(object, objectKey, resourceChangeByKarmada, propagationPolicy)
+		return d.ApplyPolicy(object, objectKey, false, propagationPolicy)
```

Comment:
/cc @RainbowMango @chaosi-zju 

My core idea is: Whether PP is Lazy or immediate, it should be immediate when the resources have not yet been bound by PP.

Reply:
I believe `Lazy` should only take effect after the PP is bound with resources.

Reply:
your this PR is the opposite of what you want.

if you mark `resourceChangeByKarmada = true`, object will always not to create the binding when `onadd` called.

Reply:
does this PR want to resolve the problem https://github.com/karmada-io/karmada/pull/4577#issuecomment-1911527287 you memtioned here?

Reply:
> I believe `Lazy` should only take effect after the PP is bound with resources.

of cource

Reply:
> does this PR want to resolve the problem https://github.com/karmada-io/karmada/pull/4577#issuecomment-1911527287 you memtioned here?


 No, these are two different issues. 

1. The issue in https://github.com/karmada-io/karmada/pull/4577#issuecomment-1911527287 is that using a single PP to select two resources with different effects.


2. This PR wants to introduce "Lazy should only take effect after the PP is bound with resources."

Reply:
Assuming that a resource has been distributed according to PolicyA.

Then PolicyA deleted, or changed `resourceSelector` and no longer match to the resource.

The detecor will remove the `policy label` in resource immediately.

This will also triggle resoure updated.

In this case, resouce will look for a new policy, if this policy is `Lazy`, the binding should be unchanged.

The related pods of the reosource running just fine, we shouldn't break it just because policy change.

Reply:
> In this case, resouce will look for a new policy, if this policy is Lazy, the binding should be unchanged.

Good question, your question introduces a new thought, whether **the binding of PP and resources** is `Lazy` or `Immediate`.

Reply:
> Lazy should only take effect after the PP is bound with resources.

This principle is what I want too, and I couldn't agree more.

However, there is an ambiguity in our understanding of `buond`. I understand `bound` to mean that a resource is bound when it is `labelled` with a policy, whereas you think that a resource is bound when it is refreshed with a binding following a policy.

Reply:
>  your question introduces a new thought, whether the binding of PP and resources is Lazy or Immediate.

So the core of my proposal is `The impact of Policy changes on the bondage relationship between resources and Policy is always immediate, but whether the resource needs to synchronize the latest placement to the binding immediately is determined by this field.`

Pay attention to `the bondage relationship between resources and Policy is always immediate`, when a resource is labelled with a policy, just finish label, it finished the `bound`.

after `bound`, there's only ever one definitive policy for this resource, if this policy is `lazy`, don't change the resource's binding.

This would be the clearest behaviour.

Reply:
> Pay attention to the bondage relationship between resources and Policy is always immediate, when a resource is labelled with a policy, just finish label, it finished the bound.


I got you.
However, most resources, once created, will not be changed.


As i said: 
> then resources like GatewayClass, IPPool can never be propagated to member clusters. Because once they are created, they will never be modified.